BackendServer/python/analysis.py

```diff
@@ -0,0 +1,163 @@
+import pandas as pd
+import numpy as np
+import mysql.connector
+from enum import Enum
+
+class AutoStation(Enum):
+    noPoints = 0
+    failed = 1
+    docked = 2
+    engaged = 3
+
+
+class TeleStation(Enum):
+    noPoints = 0
+    parked = 1
+    failed = 2
+    docked = 3
+    engaged = 4
+
+AUTO_ATTEMPTED_DOCK_STATES = (AutoStation.failed, AutoStation.docked, AutoStation.engaged)
+AUTO_SUCCESSFUL_DOCK_STATES = (AutoStation.docked, AutoStation.engaged)
+
+TELE_ATTEMPTED_DOCK_STATES = (TeleStation.failed, TeleStation.docked, TeleStation.engaged)
+TELE_SUCCESSFUL_DOCK_STATES = (TeleStation.docked, TeleStation.engaged)
+    
+# Safe division function to return None if division by zero so that it becomes NULL in the database
+def div(a, b):
+    if b == 0:
+        return None
```

Comment:
might be helpful for debugging if you add a logger or print statement here if we ever do end up dividing by zero. Adding in the basic python logger (https://docs.python.org/3/library/logging.html) or absl if you wanted to add more cool features (https://abseil.io/) could be really cool and add value for the future. This certainly doesn't need to be a right now problem, but would be cool to do in the future